src/components/Drawer/components/Drawer.tsx

```diff
@@ -0,0 +1,232 @@
+    disableBodyScrollLock = false,
+    disableVisuallyHiddenDismiss,
+    onTransitionIn,
+    onTransitionInComplete,
+    onTransitionOut,
+    onTransitionOutComplete,
+    floatingRef,
+    disablePortal,
+    hideVeil,
+    keepMounted = false,
+    container,
+    showInitialAnimation = false,
+}: DrawerProps) => {
+    const floatingNodeId = useFloatingNodeId();
+
+    const handleOpenChange = React.useCallback(
+        (isOpen: boolean, event?: Event, reason?: OpenChangeReason) => {
+            onOpenChange?.(isOpen, event, reason);
+        },
+        [onOpenChange],
+    );
+
+    const {refs, context} = useFloating({
+        nodeId: floatingNodeId,
+        open,
+        onOpenChange: handleOpenChange,
+    });
+
+    const {isInitialRender, isMounted, status, isTransitionInProgress} = useDrawerFloating({
+        onTransitionIn,
+        onTransitionInComplete,
+        onTransitionOut,
+        onTransitionOutComplete,
+        context,
+    });
+
+    const dismiss = useDismiss(context, {
+        enabled: !disableEscapeKeyDown,
+        outsidePress: false,
+        escapeKey: !disableEscapeKeyDown,
+    });
+
+    const role = useRole(context, {role: 'dialog'});
+    const {getFloatingProps} = useInteractions([dismiss, role]);
+    const veilRef = React.useRef<HTMLDivElement>(null);
+    const handleFloatingRef = useForkRef<HTMLDivElement>(refs.setFloating, floatingRef);
+
+    const handleVeilClick = React.useCallback(() => {
+        if (disableOutsideClick) {
+            return;
+        }
+
+        onOpenChange?.(false);
+    }, [disableOutsideClick, onOpenChange]);
+
+    const currentStyle = React.useMemo(() => {
+        const positionProp = {
+            position: (disablePortal ? 'absolute' : 'fixed') as 'fixed' | 'absolute',
+        };
+
+        return {
+            ...positionProp,
+            ...style,
+        };
+    }, [style, disablePortal]);
+
+    const portal =
+        isMounted || keepMounted ? (
+            <Portal container={container} disablePortal={disablePortal}>
+                <FloatingOverlay
+                    style={currentStyle}
+                    aria-modal="true"
+                    className={b(
+                        {
+                            open,
+                            'hide-veil': hideVeil,
+                            'skip-animation': !showInitialAnimation && isInitialRender,
+                        },
+                        className,
+                    )}
+                    data-qa={qa}
+                    data-floating-ui-status={status}
+                    data-transiting={isTransitionInProgress}
+                    lockScroll={!disableBodyScrollLock}
+                >
+                    <div
+                        ref={veilRef}
+                        className={b('veil', {hidden: hideVeil}, veilClassName)}
+                        role="presentation"
+                        onClick={handleVeilClick}
+                    />
```

Comment:
Why it's not like in the Modal, where veil is the root node? Seems it would be cleaner